src/shogun/clustering/KMeansBase.cpp

```diff
@@ -321,7 +321,7 @@ void KMeansBase::init()
 	SG_ADD(
 	    &use_kmeanspp, "kmeanspp", "Whether to use kmeans++",
 	    ParameterProperties::HYPER | ParameterProperties::SETTING);
-	SG_ADD(&mus, "mus", "Cluster centers", ParameterProperties::MODEL);
+	SG_ADD(&mus, "cluster_centers", "Cluster centers", ParameterProperties::MODEL);
```

Comment:
this wont work. if you rename the variable itself (mus -> cluster_centers) then you have to do it here as well.....

hence my previous suggestion that you do not need to rename any of the variable names, just how it's being registered here, and how it is called in the observe methods.

i'm pretty sure that this does not compile locally... so plz before you send in a patch make sure that it compiles... and then you'll see what are the errors you need to fix.

Reply:
@vigsterkr I am planning to rename the variables as mus -> cluster_centers and mus_initial -> initial_centers. Because at the end we want the name of the param **mus** to be changed right? And then I will register them and make changes in how they are called in the observe methods. Not sure, is it a good idea?